core/src/main/java/com/yahoo/oak/BasicChunk.java

```diff
@@ -21,27 +21,34 @@
 
 
     /*-------------- Members --------------*/
-    // to compare serilized and object keys
+    protected final OakSharedConfig<K, V> config;
+
+    // to compare serialized and object keys
     protected OakComparator<K> comparator;
+
     // in split/compact process, represents parent of split (can be null!)
     private final AtomicReference<BasicChunk<K, V>> creator;
     // chunk can be in the following states: normal, frozen or infant(has a creator)
     private final AtomicReference<State> state;
     private final AtomicReference<Rebalancer<K, V>> rebalancer;
     private final AtomicInteger pendingOps;
-    private final int maxItems;
+    protected final int maxItems;
```

Comment:
Why not to use getMaxItems()?

Reply:
It seems more natural to me to access internal *final* fields directly. I don't see why we need a function call here.
For the same reason, we access `externalSize` and `statistics` directly.